### ISSUE TITLE
如何用自己的数据集构建RQVAE呢？

### ISSUE THREAD

**Issue by wanghao19970205** [RUCAIBox/LC-Rec#9]
你好，我希望用自己的数据集，开始生成embedding以及构建RQVAE跑出来index数据，生成embedding的参考代码是不是缺一些python模块呀，比如load_plm()，我看现在的代码结构并没有这个方法。

**Comment by wanghao19970205** [comment]
我用以下代码生成embedding.npy文件，其中我一个embedding是[1, 4096]的list，总共有1000个item的embedding，经过这个代码得到张量堆叠后输出的shape是[4096000，]，这个是所需的格式吗？是否应该输出的shape为[1000, 4096]这样才是对的？
import numpy as np
import argparse
import json
import torch

def parse_args():
    parser = argparse.ArgumentParser(description="embedding to npy")

    parser.add_argument('--source_path', type=str, default="", help='ebd path')
    parser.add_argument('--save_path', type=str, default="", help='output path')
    return parser.parse_args()

def get_npyebd(args):
    origin_path = args.source_path
    embeddings = []
    with open(origin_path, "r") as f:
        for line in f:
            embedding = json.loads(line.strip())["embedding"]
            # nid = json.loads(line.strip())["nid"]
            tensor_embedding = torch.tensor(embedding)
            field_mean_embedding = torch.stack([tensor_embedding], dim=0).mean(dim=0)
            embeddings.append(field_mean_embedding)
        print(len(embeddings))
        embeddings = torch.cat(embeddings, dim=0).numpy()
        print(embeddings.shape)
        np.save(args.save_path, embeddings)
            

if __name__ == '__main__':
    args = parse_args()
    get_npyebd(args)

**Comment by zhengbw0324** [comment]
@wanghao19970205 
您好`load_plm`在[这里](https://github.com/RUCAIBox/LC-Rec/blob/4dde5da38ef67180330b773c4890fe75e78f8524/data_process/utils.py#L82)。embedding shape应该是[item_num, llm_hidden_size]，也就是[1000, 4096]。

**Comment by wanghao19970205** [comment]
好的，感谢

**Comment by jkdwxm** [comment]
请问RQ-VAE参数怎么设置呀，我也尝试在其他数据集上进行训练，但很难取得有效的性能，感觉RQ-VAE这块的训练很难控制（尤其是在总体训练时间如此长的情况下）
